pkg/util/server/recovery.go

```diff
@@ -52,11 +54,27 @@ var (
 	})
 )
 
-func onPanic(p interface{}) error {
+func onPanic(ctx context.Context, p interface{}) error {
 	stack := make([]byte, maxStacksize)
 	stack = stack[:runtime.Stack(stack, true)]
+	traceID := jaegerTraceID(ctx)
+
 	// keep a multiline stack
-	fmt.Fprintf(os.Stderr, "panic: %v\n%s", p, stack)
+	fmt.Fprintf(os.Stderr, "panic: %v %s\n%s", p, traceID, stack)
```

Comment:
I'd change it to
```suggestion
	fmt.Fprintf(os.Stderr, "panic=%v %s\n%s", p, traceID, stack)
```

but am afraid that this would break some alerts.

Reply:
You good do also:
```go
fmt.Fprintf(os.Stderr, `panic:"Look into panic=" panic=%v %s\n%s`, p, traceID, stack)
```

Reply:
Some logfmt might not be a bad idea. This looks like the best of both worlds. 
`fmt.Fprintf(os.Stderr, `panic: reason=%s traceID=%s\n%s`, p, traceID, stack)`

Or maybe drop the `reason=` if we want to maintain a traditional panic message. Adding `traceid=` seems useful though, otherwise its hard to know what the random string of characters represents.